src/scalar/mod.rs

```diff
@@ -115,7 +115,7 @@ pub fn new_scalar(array: &dyn Array, index: usize) -> Box<dyn Scalar> {
         }
         FixedSizeBinary => todo!(),
         FixedSizeList => todo!(),
-        Union => todo!(),
+        Union | Map => todo!(),
```

Comment:
I was working on map support, but didn't realise that I could skip a scalar as a start, so I got stuck here and didn't have a lot of time to continue.

I tried making a scalar to implement `HashMap` as a start, to validate that keys are hashable and unique.

Reply:
yeah, the scalar API is always a bit behind: as long as we have the iterator API to iterate over individual elements, we are good to go.